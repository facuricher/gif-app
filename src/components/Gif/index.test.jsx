import { render, screen } from "@testing-library/react"
import { GifsItem } from "."


describe('test GifsItem', () => {

  const title = 'Patricio';
  const url = 'https://bob-esponja.com/patricio.jpng';

  test('must match the snapshot', () => {
   const { container } = render(<GifsItem title={ title } url={ url } />);
   expect( container ).toMatchSnapshot();
  })

  test('must show the image and the indicated url', () => {
    render(<GifsItem title={ title } url={ url }/>);
    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url );
    expect( alt ).toBe( title );
   })

   test('that the title exists', () => {
    render(<GifsItem title={ title } url={ url }/>);

    expect( screen.getByText( title )).toBeTruthy();
   })
})
