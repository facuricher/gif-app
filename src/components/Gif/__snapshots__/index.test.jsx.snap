// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifsItem must match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Patricio"
      src="https://bob-esponja.com/patricio.jpng"
    />
    <p>
      Patricio
    </p>
  </div>
</div>
`;
